### DISCUSSION TITLE
Faces, Map View on photos from partner

### DISCUSSION THREAD

**Discussion by Th0mC** [immich-app/immich#6339] [Q&A]
Hi all,

New immich user here. Amazed by the work already done but still defining what could be missing to use it as my main photo library. And I have some doubt if I'm missing something on the following topics : 

We are currently sharing our libraries with my partner.
Plus, some photos have been added to an album I shared.

In both cases, I have the following behavior : 
I can see the photos from my partner in my "photos" feed. Good!
Opening the details, I can see the GPS metadata. Good! But : 

1. I can't edit the GPS metadata
2. In the Map View, my partner's photos are not listed
3. There is no Faces on any of my partner's photos
Neither the ones calculated in their library, neither ones that could be calculated in my library.

On the contrary, for all the photos added in my shared album by anonymous user (through shared link),
- GPS Data is editable;
- they are visible in the map view;
- and have faces calculated on them.

Is this normal behavior or I'm missing something?

**Comment by CORAAL** [comment]
Feel free to correct me if I say something wrong. 

1. To modify images (GPS...) you must be the owner of the photos. 
If your partner is the one who shared the photos with you, only he/she can edit them. 

2. (& 3.) Normal behavior. At the moment I don't think (I'm not sure) that it's still possible to share faces and photos on the map (although you can indeed consult the gps data for each image as you mentioned).

**Reply by Th0mC** [reply]
Thanks for your reply.
First I just want to know if I'm facing the normal behavior or if I'm facing an issue on my setup.
1. Understood and make sense
2. Not so clear. I'm able to consult gps on shared images when available. But still not seeing them on the map
3. Not so clear. My expectation would be to either see faces calculated on my partner library (as I can see gps data), either to be able to define/calculate faces in my library related to this images

**Comment by github-cli** [comment]
I get what he is trying to do though
immich as self hosted photos server for the family, basically the family owns the photos, not one person

**Reply by Th0mC** [reply]
Yes and no ;).
First, I try to understand how it should work out of the box to be sure I'm not facing any issue on my proper setup.
Then yes, my goal to is have something shared inside my family but it is maybe too specific to my needs. But, still have the feeling that having more possibilities on images shared by anonymous people than on my partner's shared images is not the best.

**Reply by merinof** [reply]
This is a valid question. If I am sharing my pictures with somebody, I would like them to have the same view (explore, map) I have.
Partner can't create albums from my pictures, for instance.

**Comment by andreasgerstmayr** [comment]
https://github.com/immich-app/immich/pull/7065 implements a new map setting to include partner's photos on the map, available since Immich v1.95.0.

**Comment by lex-obrien** [comment]
Agree, the work that has been done is AMAZING!  I have been looking for a solution for my images, and so far this ticks all the boxes!

I am in the "testing" mode, and I have come across a similar concern on the shared of "faces" when partner asset sharing is enabled.

I would like to see a setting that allows faces to be shared and also explored as well.

Is that something that is already on the roadmap, and/or would it be something that people think would be worth exploring?

**Comment by balthazar** [comment]
I second what @lex-obrien said. Similar to how you can share the map locations, why not have some toggle for all faces in shared albums? Or one by one if that's easier?

**Comment by StoricU** [comment]
I just want to lift this question.

I’ve recently started testing Immich, and so far, the limitations around partner sharing and face recognition are significant issues. With several thousand photos each, not being able to run face recognition on photos shared by partners or, ideally, share face recognition data between us makes Immich unsuitable for our needs at the moment.

Therefore, it would be great to receive an update on the progress of improvements to partner sharing and upcoming features. This would give us a better understanding of what's coming and when we can expect these changes.

**Reply by Qhilm** [reply]
I would be interested to understand what system you use at the moment? Is there a software that does this, sharing facial recognition date across accounts?

**Reply by rabbitambulance** [reply]
Google Photos does it. If my wife shares a pic of our son with me, that picture show up in facial recognition searches that I run. Not so with Immich.

**Reply by Qhilm** [reply]
so if your wife labelled the face "our_kid", this label will show up in your list of faces in google photos?

Or is it rather this:
1. Your wife have a face/tag called "our_kid"
2. You have a face/tag called "our_boy"
3. Wife shares picture with you
4. Google photo runs face recognition as usual on new picture and it matches your face/tag "our_boy".

In otherwords: does "normal" face recognition runs on shared pictures in google photo or does it go further and shares face/tags?

**Comment by jazmit** [comment]
+1 for sharing of faces between partners (or at least allowing running face detection on partner photos)
